test/boolean_test.cpp

```diff
@@ -505,3 +505,49 @@ TEST(Boolean, BatchBoolean) {
   EXPECT_FLOAT_EQ(subtract.Volume(), 7226.043);
   EXPECT_FLOAT_EQ(subtract.SurfaceArea(), 14904.597);
 }
+
+TEST(Boolean, NastyGears) {
+  // Create a nasty gear pattern with many rotated cubes that creates
+  // antiparallel slivers (triangles with normals ~180 degrees apart)
+  // https://github.com/BrunoLevy/thingiCSG/blob/main/DATABASE/Basic/nasty_gear_1.scad
+
+  const int N = 50;  // Number of rotations for the gear pattern
+  const double alpha = 360.0 / N;
+
+  // Create outer gear - many rotated cubes unioned together
+  std::vector<Manifold> outerCubes;
+  for (int i = 0; i < N; ++i) {
+    outerCubes.push_back(
+        Manifold::Cube({1, 1, 1}, true).Rotate(0, 0, alpha * i));
```

Comment:
This isn't introducing any rounding error in Z, so I'm a little surprised we're having trouble with slivers. Seems like symbolic perturbation has failed - perhaps we should debug that first. Does it also repro with N = 2 or 4? Also, have you tried forcing evaluation of the gear, before you subtract them from eachother, e.g. `OuterGeart.NumVert()`? I'm curious if the Boolean reordering is giving us trouble.

Reply:
I think @pca006132 once figured out how to get triangulation errors / slivers to arise with just two strategically placed rotated cubes, but I think it’s non-deterministic, so a high N is just creating many chances for it to occur (since coplanar simplification is also having trouble, so the top and bottom surfaces are quite crunchy).

I’ll report back soon with more detailed pictures of what happens when N is low or the gear is pre-evaluated…

Reply:
> Does it also repro with N = 2 or 4? Also, have you tried forcing evaluation of the gear, before you subtract them from eachother, e.g. `OuterGeart.NumVert()`? I'm curious if the Boolean reordering is giving us trouble.

Slivers don't repro until N = 10.  Also, I was wrong earlier, these slivers are deterministic.

Forcing evaluation of the gear before subtraction doesn't change the number or location of the slivers :-(